Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -18,26 +18,69 @@
 public interface SeekableChannelsProvider extends SafeCloseable {
 
     /**
-     * Take the file source path or URI and convert it to a URI object.
+     * Take the file source path or URI string and convert it to a URI object.
      *
      * @param source The file source path or URI
      * @return The URI object
      */
     static URI convertToURI(final String source) {
+        if (source.isEmpty()) {
+            throw new IllegalArgumentException("Cannot convert empty source to URI");
+        }
         final URI uri;
         try {
             uri = new URI(source);
         } catch (final URISyntaxException e) {
             // If the URI is invalid, assume it's a file path
-            return new File(source).toURI();
+            return convertFileToURI(new File(source));
         }
         if (uri.getScheme() == null) {
-            // Need to convert to a "file" URI
-            return new File(source).toURI();
+            // Convert to a "file" URI
+            return convertFileToURI(new File(source));
         }
         return uri;
     }
 
+    /**
+     * Takes a file and convert it to a URI object. We should use this method instead of {@link File#toURI()} because
+     * {@link File#toURI()} internally calls {@link File#isDirectory()}, which can lead to disk access.
+     *
+     * @param file The file
+     * @return The URI object
+     */
+    static URI convertFileToURI(final File file) {
```

Comment:
The file.toURI() method does a isDirectory call and adds a slash at the end of the URI if the file is a directory.

I skipped that check completely because all the tests (both local and S3) passed without it. I also verified reading flat partitioned and key-value partitioned data from S3 and that also worked with URIs which correspond to directories but are not followed by a slash.

Reply:
I somewhat like a single method w/ boolean: `convertFileToURI(final File file, boolean isDirectory)`. In this way we can claim to preserve the semantics of `File#toURI` by letting the caller know they should have knowledge of the type of file. Are there any cases we have where the callsite doesn't know whether it's supposed to be a file or directory?

Reply:
There are some such cases, like when user calls a `readTable(String)`, then we don't know if String is a file or a directory. I am adding a basic check method in that case.